src/main/java/no/fdk/dataservicecatalog/service/DataServiceService.java

```diff
@@ -107,7 +110,7 @@ private void triggerHarvest(DataService dataService) {
         return dataServiceMongoRepository.save(dataService)
                 .doOnSuccess(saved -> {
                     log.debug("dataservice {} saved", saved.getId());
-                    if (saved.getStatus() != null && saved.getStatus().equals(Status.PUBLISHED)) {
+                    if (saved.getStatus().equals(Status.PUBLISHED)) {
```

Comment:
Removing null check will throw a NullPointerException (see link below). If you want shorter syntax, use == instead.

https://stackoverflow.com/questions/1750435/comparing-java-enum-members-or-equals